OPHD/UI/Reports/ResearchReport.cpp

```diff
@@ -95,37 +100,84 @@ ResearchReport::~ResearchReport()
 
 void ResearchReport::fillLists()
 {
+	lstResearchTopics.clear();
+	resetCategorySelection();
+	onAllTopicsClicked();
 }
 
 
 void ResearchReport::clearSelected()
 {
+	lstResearchTopics.clearSelected();
+	resetCategorySelection();
+	onAllTopicsClicked();
 }
 
 
 void ResearchReport::refresh()
 {
-	if (CategoryPanels.size() < 1) { return; }
+	if (mCategoryPanels.size() < 1) { return; }
+
+	adjustCategoryIconSpacing();
+
+	resetCategorySelection();
+	onAllTopicsClicked();
+
+	setSectionRects();
+
+	lstResearchTopics.area(mResearchTopicArea);
+}
+
+
+void ResearchReport::setSectionRects()
+{
+	mCategoryIconArea =
+	{
```

Comment:
This is a function with both a significantly complex calculation, and a side effect (assigning to member variables). It may be wise to separate the two actions, particularly if you want to add tests for the calculation at some point.

Instead, I would suggest calculation functions that return the resulting value, which would be separate from any function that might call them and assign the result to member variables.

----

If the write is the only access to member variables, than a separated calculation function could be left at module level scope (as opposed to making it a `static` member function). That limits visibility since it no longer needs to be declared in a header file.

Reply:
Totally agreed, something I plan to do in a later change set.